### ISSUE TITLE
Errors can fail silently

### ISSUE THREAD

**Issue by ASLeonard** [malonge/RagTag#31]
Hi,
I've noticed a few times that, at least on cluster submitted jobs, errors during the scaffolding process don't seem to cause RagTag to fail. Specifically, I am using RagTag as a step in a snakemake workflow, and everything appears to have gone smoothly and continue, when in reality there was an error.

For example, a stripped down log is below
```
Fri Oct 23 20:31:14 2020 --- Writing: 
Fri Oct 23 20:31:14 2020 --- Retaining pre-existing file: 
Fri Oct 23 20:31:14 2020 --- Running: ragtag_agp2fasta.py 
[E::fai_retrieve] Failed to retrieve block: unexpected end of file
Traceback (most recent call last):
...
 File "pysam/libcfaidx.pyx", line 319, in pysam.libcfaidx.FastaFile.fetch
ValueError: failure when retrieving sequence on 'tig00018983'
Traceback (most recent call last):
...
RuntimeError: Failed : ragtag_agp2fasta.py 
```

I'm not sure why the RuntimeError isn't signalling correctly that there has been a failure, maybe something to do with how `run_o` redirects stdout? However, since the file is progressively generated, there is an output file (just very incomplete) which snakemake believes is correct and moves on. 

I believe this specific issue is due re-using the .agp file with a modified input, which wouldn't occur if I used `-w`, so that doesn't seem troublesome in general.

Thanks,
Alex

**Comment by malonge** [comment]
Hi there,

Thanks for bringing this up and I apologize for the delayed response. This is definitely concerning, though I am not quite sure how I would go about debugging this. I will leave the issue open to see if others experience this or have insight. 

Thanks,
Mike

**Comment by ASLeonard** [comment]
It appears this was fixed in bc54e9d7bb, as I'm pretty sure it was when the agp file and the faidx file were out of sync (for me it was faidx updated on a newer assembly but the agp was retained).

**Comment by malonge** [comment]
Great - glad this got resolved. I was ultimately able to make this change because now users can use `agp2fasta` if they want to make custom AGPs and then convert those to FASTAs.
